extensions/amp-story-shopping/0.1/amp-story-shopping-tag.js

```diff
@@ -12,19 +16,46 @@ export class AmpStoryShoppingTag extends AMP.BaseElement {
 
     /** @private {?Element} */
     this.container_ = null;
+
+    /** @private @const {!./amp-story-store-service.AmpStoryStoreService} */
+    this.storeService_ = null;
   }
 
   /** @override */
   buildCallback() {
     super.buildCallback();
     this.container_ = this.element.ownerDocument.createElement('div');
     this.container_.textContent = this.myText_;
+    this.container_.id = this.element.id;
     this.element.appendChild(this.container_);
     applyFillContent(this.container_, /* replacedContent */ true);
+
+    return Promise.all([
+      Services.storyStoreServiceForOrNull(this.win).then((storeService) => {
+        this.storeService_ = storeService;
+        this.storeService_.subscribe(
+          StateProperty.SHOPPING_STATE,
+          (shoppingState) => {
+            this.updateShoppingTag_(shoppingState);
+          }
+        );
+      }),
+    ]);
   }
 
   /** @override */
   isLayoutSupported(layout) {
     return layout === Layout.CONTAINER;
   }
+
+  /**
+   * @param {Object} shoppingState
+   * @private
+   */
+  updateShoppingTag_(shoppingState) {
+    if (shoppingState[this.container_.id] != null) {
+      this.myText_ = shoppingState[this.container_.id]['product-title'];
+      this.container_.textContent = this.myText_;
```

Comment:
`shoppingState[this.container_.id]` is accessed twice, lets make it a variable and name it so this is more readable. Maybe it's something like:

```
const tagData = data[this.element.dataset.tagId];
if (tagData) {
  this.element.textContent = tagData['product-title']
}
```